uxarray/grid.py

```diff
@@ -540,6 +542,73 @@ def _build_edge_node_connectivity(self, repopulate=False):
         setattr(self, "Mesh2_edge_nodes", self.ds['Mesh2_edge_nodes'])
         setattr(self, "nMesh2_edge", edge_nodes_unique.shape[0])
 
+    def _build_node_faces_connectivity(self):
+        """Builds the `Grid.Mesh2_node_faces`: integer DataArray of size
+        (nMesh2_node, MaxNumFacesPerNode) (optional) A DataArray of indices
+        indicating faces that are neighboring each node.
+
+        This function converts the face-node connectivity data into a sparse matrix, and then constructs the node-face
+        connectivity by iterating over each node in the mesh and retrieving the set of neighboring faces.
+
+        Raises
+        ------
+        RuntimeError
+            If the Mesh object does not contain a 'Mesh2_face_nodes' variable.
+        """
+        if "Mesh2_face_nodes" not in self.ds:
+            raise RuntimeError(
+                "Currently, node_face_connectivity can only be built with Mesh2_face_nodes"
+            )
```

Comment:
I believe that we will always have `Mesh2_face_nodes`, we should keep these types of checks outside these build functions. As we discussed in a previous PR, we'll have separate functions exposed to the user that will handle these checks.

Reply:
we can delete it later after we have that checking function working. Currently, I don't feel comfortable exposing the API without any pre-check.

Reply:
This works with me